src/main/java/eggventory/model/StateInterface.java

```diff
@@ -0,0 +1,146 @@
+        historyList.pushLoanListSave(loanList.saveLoanListString());
+        historyList.pushPersonListSave(personList.savePersonListString());
+        historyList.pushTemplateListSave(templateList.saveTemplateListString());
+    }
+
+    /**
+     * Pushes state of model components converted into String to futurelist.
+     */
+    public void pushStateFutureList() {
+        StockList stockList = currentState.getStockList();
+        LoanList loanList = currentState.getLoanList();
+        PersonList personList = currentState.getPersonList();
+        TemplateList templateList = currentState.getTemplateList();
+
+        //Push current state into futurelist
+        futureList.pushStockSave(stockList.saveDetailsString());
+        futureList.pushStockTypeSave(stockList.saveStockTypesString());
+        futureList.pushLoanListSave(loanList.saveLoanListString());
+        futureList.pushPersonListSave(personList.savePersonListString());
+        futureList.pushTemplateListSave(templateList.saveTemplateListString());
+    }
+
+
+    public StockList getStockList() {
+        return currentState.getStockList();
+    }
+
+    public LoanList getLoanList() {
+        return currentState.getLoanList();
+    }
+
+    public PersonList getPersonList() {
+        return currentState.getPersonList();
+    }
+
+    public TemplateList getTemplateList() {
+        return currentState.getTemplateList();
+    }
+
+    /**
+     * Pushes current state into futurelist and loads last state from historylist.
+     * @throws BadInputException when previous state contains bad input.
+     */
+    public void executeUndoCommand() throws BadInputException {
+        if (historyList.isEmpty()) {
+            throw new BadInputException("The UNDO command could not be executed.\n");
+        }
+
+        //Push current state information into futurelist and clear current static state variables
+        pushStateFutureList();
+        getStockList().getList().clear();
+        getPersonList().getPersonList().clear();
+        getTemplateList().getTemplates().clear();
+        getLoanList().getLoansList().clear();
+
+        //Load the newstate information from history list
+        StockList updatedStockList = new StorageStub().loadStockList(historyList.popStockSave(),
+                historyList.popStockTypeSave());
+        LoanList updatedLoanList = new StorageStub().loadLoanList(historyList.popLoanListSave());
+        PersonList updatedPersonList = new StorageStub().loadPersonList(historyList.popPersonListSave());
+        TemplateList updatedTemplateList = new StorageStub().loadTemplateList(historyList.popTemplateListSave());
+
+
+        //Update currentstate infromation with new state information
+        currentState.setStockList(updatedStockList);
+        currentState.setLoanList(updatedLoanList);
+        currentState.setPersonList(updatedPersonList);
+        currentState.setTemplateList(updatedTemplateList);
```

Comment:
Loading and updating of all states could be done in private methods. LoadAllStates(), UpdateAllStates()

Reply:
Same with clear

Reply:
Alright, thanks, Time to OOPIFY

Reply:
managed to OOPiFY the clear and setAllState. But the loader I could not, because of the static issue. gonna move on for now, will get back to it if I have time